src/esrs/state.rs

```diff
@@ -10,48 +11,49 @@ use crate::types::SequenceNumber;
 /// - a lock representing the atomicity of the access to the aggregate,
 /// - a state defined by the user of this library.
 pub struct AggregateState<S> {
-    pub(crate) id: Uuid,
-    pub(crate) sequence_number: SequenceNumber,
-    pub(crate) lock: Option<EventStoreLockGuard>,
-    pub(crate) inner: S,
-}
-
-impl<S: Clone> Clone for AggregateState<S> {
-    fn clone(&self) -> Self {
-        Self {
-            id: self.id,
-            sequence_number: self.sequence_number,
-            inner: self.inner.clone(),
-            lock: None,
-        }
-    }
+    id: Uuid,
+    sequence_number: SequenceNumber,
+    lock: Option<EventStoreLockGuard>,
+    inner: S,
 }
 
 impl<S: std::fmt::Debug> std::fmt::Debug for AggregateState<S> {
     fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
         f.debug_struct("AggregateState")
             .field("id", &self.id)
             .field("sequence_number", &self.sequence_number)
-            .field("inner", &self.inner)
             .field("lock", &self.lock.is_some())
+            .field("inner", &self.inner)
             .finish()
     }
 }
 
 /// Default implementation for [`AggregateState`]
 impl<S: Default> Default for AggregateState<S> {
     fn default() -> Self {
-        Self::new(Uuid::new_v4())
+        Self::new()
     }
 }
 
 impl<S: Default> AggregateState<S> {
-    /// Creates a new instance of an [`AggregateState`] with the given aggregate id. The use of this
-    /// is discouraged being that that aggregate id could be already existing and a clash of ids
-    /// might happen.
+    /// Creates a new instance of an [`AggregateState`] with a new unique id.
+    pub fn new() -> Self {
+        Self {
+            id: Uuid::new_v4(),
+            inner: Default::default(),
+            sequence_number: 0,
+            lock: None,
+        }
+    }
+
+    /// Creates a new instance of an [`AggregateState`] with the given aggregate id.
     ///
-    /// Prefer [Default] implementation.
-    pub fn new(id: impl Into<Uuid>) -> Self {
+    /// This should be used almost exclusively when loading by aggregate id yields nothing,
+    /// and this becomes the brand new aggregate state for that id.
+    ///
+    /// Other uses are strongly discouraged, since the id could be already existing
+    /// and a clash might happen when persisting events.
+    pub fn with_id(id: impl Into<Uuid>) -> Self {
```

Comment:
Turns out there is no way out of this, we NEED a constructor with id (unless we expose the internal `id` field). I went with a less suggestive/more specific name instead.